src/routes/preHandlers/users.js

```diff
@@ -217,6 +217,7 @@ exports.authorizeLearnersGet = async (req) => {
   if (!vendorRole && (!isClientRoleAllowed || !isQueryCompanyValid)) throw Boom.forbidden();
   if (!clientRole && !isVendorRoleAllowed) throw Boom.forbidden();
   if (!isClientRoleAllowed && !isVendorRoleAllowed) throw Boom.forbidden();
+  if (query.hasCompany && (!vendorRole || query.company)) throw Boom.forbidden();
```

Comment:
Je ne sais pas si il faut bloquer le hasCompany si il y a une entreprise en query. Ce n'est pas genant dans le sens ou ca va juste verifier que company existe bien en plus d'etre egale a l'id voulu. Qu'en pensez-vous ?

Reply:
ouais mais du coup ça casse complètement ton helper, du coup tu fais comment tu changes le helper pour qu'il puisse à la fois recevoir les 2 query ? ça complique pas un peu la chose inutilement ?

Reply:
Je suis assez d'accord avec Ulysse, on ne devrait pas avoir de cas en front où l'on peut envoyer les deux queries (hasCompany et company) et ça complique le code si jamais on laisse passer le cas où il y a les deux.

Reply:
Je me demande d'ailleurs si avec un Joi.alternatives ou un max(1) dans validate ça ne serait pas plus compréhensible.
Qu'en pensez-vous ?

Reply:
J'avais l'impression que justement ca ne cassait pas le helper d'ou mon interrogation, mais si ca casse le helper vaut mieux laisser en effet. 
Je suis d'accord avec la suggestion de Kenny pour Joi.alternatives :)

Reply:
je ne suis pas sur de comprendre ce qus vous dites : en fait pour moi, le cas `company` implique le cas `hasCompany` donc je n'ai pas l'impression que ca complique le code et pas besoin de gerer le cas ou il recoit les 2 query
en terme de logique c'est un ET, si tu filtres les utilisateur qui ont une structure ET qui sont rattachés a l'ASAPAD, c'est pareil que de filtrer les utilisateurs rattachés a l'ASAPAD

Reply:
@sophiemoustard cas où on appelle avec juste company: this.selectedCompany  dans le helper le userQuery = company: '5ed671c6ddd9f3001495c1dc',
  'role.client': { '$not': { '$in': [Array] } }
}

cas où on appelle avec company: this.selectedCompany, hasCompany: true dans le helper le userQuery =   company: { '$exists': true },
  'role.client': { '$not': { '$in': [Array] } }
}

on a plus l'id de la company et on renvoie pas du tout ce qu'on voulait à la base (d'ailleurs ça pète en front du coup)

donc en soit on peut complexifier le code pour prendre en compte le cas où on reçoit les 2 (mais ça me semble un peu complexifier la chose pour rien) ou laisser comme c'était avant, mais en tout cas on peut pas retirer le if sans rien changer dans le helper

Reply:
pour le alternatives on peut l'utiliser oui ! mais je pense qu'il faut toujours verifier le role vendeur non ?